tests/test_abstractions.py

```diff
@@ -1,302 +1,69 @@
+from transformer_lens import HookedTransformer
+
+
+@pytest.fixture(scope="module")
+def transformer():
+    # No strong reason this needs to be a HookedTransformer, just means we don't have
+    # to explicitly deal with the tokenizer later.
+    # model = HookedTransformer.from_pretrained_no_processing(
+    #     "attn-only-1l", device="cpu"
+    # )
+    official_model_name = loading.get_official_model_name("attn-only-1l")
+    cfg = loading.get_pretrained_model_config(
+        official_model_name, fold_ln=False, device="cpu"
     )
-    expected_dims = [1, 2, 2, 3, 6]
-
-    @abstractproperty
-    def get_default_abstraction(
-        self,
-    ) -> Callable[[models.HookedModel, int], Abstraction]:
-        pass
-
-    def test_default_abstraction_shapes(self):
-        abstraction = self.get_default_abstraction(self.model, size_reduction=2)
-        assert len(abstraction.tau_maps) == len(self.expected_dims)
-        for i in range(len(self.expected_dims)):
-            tau_map = abstraction.tau_maps[f"post_linear_{i}"]
-            if i == len(self.expected_dims) - 1:
-                assert isinstance(tau_map, nn.Identity)
-            else:
-                assert isinstance(tau_map, nn.Linear)
-                assert tau_map.in_features == self.full_dims[i]
-                assert tau_map.out_features == self.expected_dims[i]
-
-    def test_default_abstraction_forward_pass(self):
-        abstraction = self.get_default_abstraction(self.model, size_reduction=2)
-        inputs = torch.randn(1, 28, 28)
-        names = [f"post_linear_{i}" for i in range(len(self.expected_dims))]
-        output, activations = self.model.get_activations(inputs, names=names)
-        assert len(activations) == len(self.expected_dims)
-        for name, activation in activations.items():
-            assert name.startswith("post_linear_")
-            assert name.split("_")[-1].isdigit()
-            assert activation.ndim == 2
-            assert activation.shape[0] == 1
-            assert activation.shape[1] == self.full_dims[int(name.split("_")[-1])]
-
-        abstractions, _ = abstraction(activations)
-        assert abstractions.keys() == _.keys()
-        assert abstractions.keys() == activations.keys()
-        for k, v in abstractions.items():
-            n = int(k[-1])
-            assert v.ndim == 2
-            assert v.shape[0] == 1
-            assert v.shape[1] == self.expected_dims[n]
-
-
-class ABCTestCNNAbstraction(ABC):
-    cnn_dims = [2, 3, 4, 5]
-    mlp_dims = [7, 11, 10]
-    model = models.CNN(
-        input_shape=(1, 28, 28),
-        channels=cnn_dims,
-        output_dim=mlp_dims[-1],
-        dense_dims=mlp_dims[:-1],
+    model = HookedTransformer(cfg)
+    model.init_weights()
+    return model
+
+
+@pytest.fixture
+def text():
+    return "Hello, world!"
+
+
+@pytest.fixture
+def text_batch():
+    return ["Hello, world!", "This is a test."]
+
+
+def test_lca_identity_abstraction(transformer, text_batch):
```

Comment:
We could consider a mark for tests that has transformer_lens dependencies and move the import inside the `def transformer():` function to be able to run the rest of tests without the extra dependency. But, it's okay as is.

Reply:
Oh, but all tests in this file has that dependency. Then no mark is needed.